### PR TITLE
[B2BP-171] - Terraform DB ALB ECS Cluster AWS

### PR DESCRIPTION
**Pull request by maurodandrea**
Creation of Terraform scripts for the automatic deployment on AWS of Database, ALB and ECS Cluster resources necessary for Strapi.

Depends on https://github.com/pagopa/b2b-portals/pull/45

#### List of Changes
Creation of Terraform scripts for the automatic deployment on AWS of Database, ALB and ECS Cluster resources necessary for Strapi.

#### Motivation and Context
As reported in the [RFC "SV-003 - Architettura Strapi on AWS"](https://pagopa.atlassian.net/wiki/spaces/SV/pages/795771078/SV-003+-+Architettura+Strapi+on+AWS), to host the Strapi instance on the AWS Cloud Provider it is necessary to create the relevant resources using Terraform scripts, such as: Strapi Server (Cluster ECS Fargate), Database RDS Aurora PostgreSQL, Bucket S3 for Media Library, Load Balancer, Role IAM and Secrets SSM)

#### How Has This Been Tested?
Running Terraform scripts on locally PC pointing to a test AWS environment

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change not requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .infrastructure/98_variables.tf [hunk 1]
```diff
@@ -24,4 +24,80 @@ variable "tags" {
...
+
+variable "db_name" {
+  description = "Database Name"
+  default     = "strapidb"
+}
+
+variable "db_client" {
+  description = "Database client"
+  default     = "postgres"
+}
```
Comment (datalek) [inline .infrastructure/98_variables.tf:42]: This variable is useless

File: .infrastructure/98_variables.tf
Comment (datalek) [inline .infrastructure/98_variables.tf:1]: Almost all the variables are used only in one place. IMHO remove them and use directly the value where they are referenced

**Review by datalek — CHANGES_REQUESTED** [review]

File: .infrastructure/06_alb.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
+resource "aws_alb" "cms_load_balancer" {
+  name            = var.cms_alb
+  subnets         = module.vpc.public_subnets
+  security_groups = [aws_security_group.cms_lb.id]
+  internal        = false
+}
+
+resource "aws_alb_target_group" "app" {
```
Comment (datalek) [inline .infrastructure/06_alb.tf:8]:
```suggestion
resource "aws_alb_target_group" "cms" {
```

File: .infrastructure/06_alb.tf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    path                = var.health_check_path
+    unhealthy_threshold = "2"
+  }
+}
+
+# Redirect all traffic from the ALB to the target group
+resource "aws_alb_listener" "front_end" {
+  load_balancer_arn = aws_alb.cms_load_balancer.id
+  port              = 1337
+  protocol          = "HTTP"
```
Comment (datalek) [inline .infrastructure/06_alb.tf:30]: I'm not sure about this two lines, I think we have to set `HTTPS` and `443` port

File: .infrastructure/98_variables.tf (hunk 1 above)
Comment (maurodandrea) [reply .infrastructure/98_variables.tf:42]: variable "db_client" it is used by the ECS task definition, which is present in the next PR. If it's not a problem I would leave it defined here

File: .infrastructure/98_variables.tf
Comment (maurodandrea) [reply .infrastructure/98_variables.tf:1]: ok

File: .infrastructure/06_alb.tf (hunk 2 above)
Comment (maurodandrea) [reply .infrastructure/06_alb.tf:8]: ok

File: .infrastructure/98_variables.tf (hunk 1 above)
Comment (maurodandrea) [reply .infrastructure/98_variables.tf:42]: with next commit remove this variable (and other)

File: .infrastructure/06_alb.tf (hunk 3 above)
Comment (maurodandrea) [reply .infrastructure/06_alb.tf]: for Port=443 no problem

File: .infrastructure/06_alb.tf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  vpc_id      = module.vpc.vpc_id
+  target_type = "ip"
+
+  health_check {
+    healthy_threshold   = "3"
+    interval            = "30"
+    protocol            = "HTTP"
+    matcher             = "200"
+    timeout             = "3"
+    path                = "/"
```
Comment (datalek) [inline .infrastructure/06_alb.tf:21]: What about using [`_health` endpoint](https://forum.strapi.io/t/how-to-know-if-strapi-server-is-ready/3918/1)?

```suggestion
    matcher             = "204"
    timeout             = "3"
    path                = "/_health"
```

File: .infrastructure/06_alb.tf [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    unhealthy_threshold = "2"
+  }
+}
+
+# Redirect all traffic from the ALB to the target group
+resource "aws_alb_listener" "front_end" {
+  load_balancer_arn = aws_alb.cms_load_balancer.id
+  port              = 443
```
Comment (datalek) [inline .infrastructure/06_alb.tf:29]: The [default port is `HTTP`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_listener#protocol), we have to explicit `HTTPS`

**Review by datalek — CHANGES_REQUESTED** [review]

File: .infrastructure/06_alb.tf [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  default_action {
+    target_group_arn = aws_alb_target_group.cms.id
+    type             = "forward"
+  }
+}
+
+locals {
+  alb_origin_id = aws_alb.cms_load_balancer.id
+}
+
```
Comment (datalek) [inline .infrastructure/06_alb.tf:40]: This seems useless :)

```suggestion
```

File: .infrastructure/06_alb.tf [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    timeout             = "3"
+    path                = "/_health"
+    unhealthy_threshold = "2"
+  }
+}
+
+# Redirect all traffic from the ALB to the target group
+resource "aws_alb_listener" "front_end" {
+  load_balancer_arn = aws_alb.cms_load_balancer.id
+  port              = 443
```
Comment (datalek) [inline .infrastructure/06_alb.tf:29]: Maybe this port is not required anymore?

**Review by datalek — CHANGES_REQUESTED** [review]

File: .infrastructure/06_alb.tf (hunk 7 above)
Comment (maurodandrea) [reply .infrastructure/06_alb.tf:29]: correct

**Review by datalek — APPROVED** [review]

**Pull request merged**